streams/src/main/java/org/apache/kafka/streams/kstream/internals/FullChangeSerde.java

```diff
@@ -65,8 +71,8 @@ public void configure(final Map<String, ?> configs, final boolean isKey) {
                 final byte[] newBytes = data.newValue == null ? null : innerSerializer.serialize(topic, data.newValue);
                 final int newSize = newBytes == null ? -1 : newBytes.length;
 
-                final ByteBuffer buffer = ByteBuffer.allocate(
-                    4 + (oldSize == -1 ? 0 : oldSize) + 4 + (newSize == -1 ? 0 : newSize)
+                final ByteBuffer buffer = ByteBuffer.wrap(
```

Comment:
It's just evidence of my mental slowness...

In the prior PR, Guozhang pointed out that my calling `buffer.array()` was incorrect, since the backing array isn't guaranteed to be exactly within the bounds we allocated. I fixed it at the time by delegating to the `ByteBufferSerializer`, which handles this.

Later on I realized that there is a more efficient solution available. By pre-creating the backing array and wrapping it, we know that `buffer.array()` returns what we needed. No need for the more general handling logic in `ByteBufferSerializer`.